bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -127,6 +127,21 @@ public GarbageCollectorThread(ServerConfiguration conf, LedgerManager ledgerMana
                 Executors.newSingleThreadScheduledExecutor(new DefaultThreadFactory("GarbageCollectorThread")));
     }
 
+    private EntryLogMetadataMap createEntryLogMetadataMap(boolean gcPersistentEntrylogMetadataMapEnabled) {
+        final String baseDir = this.conf.getGcPersistentEntrylogMetadataMapPath() != null
+                ? this.conf.getGcPersistentEntrylogMetadataMapPath()
```

Comment:
i think conf.getGcPersistentEntrylogMetadataMapPath (method in ServerConfiguration.java) should handle this logic. I mean if the config value is null then use ledgerDirs().get(0).